File: python/pyfind/pyfind/option.py
# -*- coding: utf-8 -*-
"""
###############################################################################
#
# option.py
#
# class Option: abstract class that encapsulates a (command-line) option
#
###############################################################################
"""


class Option:
    """an abstract class to encapsulate a specific command line option"""
    __slots__ = ['short_arg', 'long_arg', 'desc', 'arg_type']

File: java/javafind/lib/src/main/java/javafind/Option.java
/*******************************************************************************
FindOption

Class to encapsulate a command line find option

@author Cary Clark &lt;clarkcb@gmail.com&gt;
@version $Rev$
@copyright Cary Clark 2012
*******************************************************************************/

package javafind;

public interface Option {
    String shortArg();
    String longArg();
    String description();
    ArgTokenType argType();
}
